internal/transport/http2_client.go

```diff
@@ -1255,7 +1270,11 @@ func (t *http2Client) handleGoAway(f *http2.GoAwayFrame) {
 	id := f.LastStreamID
 	if id > 0 && id%2 == 0 {
 		t.mu.Unlock()
-		t.Close(connectionErrorf(true, nil, "received goaway with non-zero even-numbered numbered stream id: %v", id))
+		err := connectionErrorf(true, nil, "received goaway with non-zero even-numbered numbered stream id: %v", id)
+		if logger.V(logLevel) {
+			logger.Errorf("transport: closing transport due to error in goaway handling: %v.", err)
+		}
```

Comment:
This would be unnecessary with a log in `t.Close`.

More below.

Reply:
Sgtm.